test/e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,397 @@
+/*
+Copyright 2024 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package controllersE2Etest
+
+import (
+	"context"
+	"log/slog"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/go-logr/logr"
+	. "github.com/onsi/ginkgo/v2/dsl/core"
+	. "github.com/onsi/gomega"
+	siteconfig "github.com/stolostron/siteconfig/api/v1alpha1"
+	admissionregistrationv1 "k8s.io/api/admissionregistration/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	clusterv1 "open-cluster-management.io/api/cluster/v1"
+	policiesv1 "open-cluster-management.io/governance-policy-propagator/api/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+
+	ibgu "github.com/openshift-kni/cluster-group-upgrades-operator/pkg/api/imagebasedgroupupgrades/v1alpha1"
+	hwv1alpha1 "github.com/openshift-kni/oran-o2ims/api/hardwaremanagement/v1alpha1"
+	provisioningv1alpha1 "github.com/openshift-kni/oran-o2ims/api/provisioning/v1alpha1"
+	provisioningcontrollers "github.com/openshift-kni/oran-o2ims/internal/controllers"
+	"github.com/openshift-kni/oran-o2ims/internal/controllers/utils"
+	testutils "github.com/openshift-kni/oran-o2ims/test/utils"
+	assistedservicev1beta1 "github.com/openshift/assisted-service/api/v1beta1"
+)
+
+const testHwMgrPluginNameSpace = "hwmgr"
+
+var (
+	K8SClient                     client.Client
+	K8SManager                    ctrl.Manager
+	ProvReqTestReconciler         *provisioningcontrollers.ProvisioningRequestReconciler
+	ClusterTemplateTestReconciler *provisioningcontrollers.ClusterTemplateReconciler
+	testEnv                       *envtest.Environment
+	ctx                           context.Context
```

Comment:
it seems like this global ctx is being overwritten in BeforeSuite and BeforeEach (with and without cancel)...might be a little unclear of the intention and potential resource leak? Might worth splitting it up? e.g in BeforeEach you have its own `testCtx := context.Background() // or context.WithTimeout if needed`